src/include/OSL/rendererservices.h

```diff
@@ -156,6 +276,26 @@ class OSLEXECPUBLIC RendererServices {
                                   Vec3* Pout, int npoints,
                                   TypeDesc::VECSEMANTICS vectype);
 
+    /// @brief Builds a free function to provide a value for a given attribute.
+    /// @param group The shader group currently requesting the attribute.
+    /// @param object_name The object name. An empty string will be provided if
+    /// the value is not specified or it is not known at compile time.
+    /// @param attribute_name The attribute name. An empty string will be
+    /// provided if the value is not known at compile time.
+    /// @param type The type of the value being requested.
+    /// @param derivatives True if derivatives are also being requested.
+    /// @param object_lookup True if an object name was specified, even if the
+    /// value is not known at compile time.
+    /// @param array_lookup True if the attribute lookup provides an index.
+    /// @param spec The built attribute getter.
+    /// @return True if an attribute builder was provided. If false, the
+    /// existing get_attribute or get_array_attribute functions will be
+    /// used as the fallback.
```

Comment:
If our intent is to get rid of the fallback get_attribute or get_array_attribute functions, then if a renderer returns false, would it be better to assume that the renderer has no intent on providing a result for the attribute_name/object_name and not generate any function calls in the JIT (saving overhead).  If the renderer wanted a fallback then they could have populated the AttributeGetterSpec with one that takes AttributeSpecBuiltinArg::ObjectName and AttributeSpecBuiltinArg::AttributeName.

And perhaps this is different than renderer choosing not to utilize the AttributeGetterSpec, maybe a ShadingSystem attribute or call to `virtual int RendererServices::supports("AttributeGetterSpec")` to identify if the JIT should even call build_attribute_getter?